workers/social/lib/social/models/group/permissionset.coffee

```diff
@@ -202,10 +208,32 @@ module.exports = class JPermissionSet extends Module
       permissions = (p.permission for p in advanced).join ', '
 
       JPermissionSet.checkPermission client, advanced, this, rest,
-        (err, hasPermission) ->
+        (err, hasPermission, permissionIndex) ->
           args = [client, rest..., callback]
           if err then callback err
           else if hasPermission
+            # client._allowedPermissionIndex
+            #
+            # if you write something like this;
+            #
+            #   foo: permit
+            #
+            #     advanced: [
+            #       { permission   : 'some permission', validateWith: Validators.own }
+            #       {
+            #         permission   : 'access related'
+            #         validateWith : accessValidator ACCESSLEVEL.READ
+            #       }
+            #       { permission   : 'modify object', superadmin: yes }
+            #     ]
+            #
+            #     success: (client, callback) ->
+            #       ...
+            #
+            # and this permission is granted because you were a superadmin then
+            # client._allowedPermissionIndex will become 2. But if you are owner of
+            # this instance this time it will become 0 because the first rule granted.
+            args[0]._allowedPermissionIndex = permissionIndex
```

Comment:
you're right with your concerns, but if this `client` will be passing around this `_allowedPermissionIndex` will be overwritten on the place last it's used. so, it will be the issue on that last method it's used. if one is using this feature inside of an insecure method it's an issue with the implementation. similar cases valid for whole permit flow btw :) 

I'm open for any alternative ways to achieve this ;)